api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java

```diff
@@ -312,6 +317,84 @@ public CriteriaListWithCountResponse findCriteriaByDomain(
         .totalCount(dbCriteriaPage.getTotalElements());
   }
 
+  @Override
+  public CriteriaListWithCountResponse findCriteriaByDomainV2(
+      String domain, String term, String surveyName, Boolean standard, Integer limit) {
+    PageRequest pageRequest =
+        PageRequest.of(0, Optional.ofNullable(limit).orElse(DEFAULT_CRITERIA_SEARCH_LIMIT));
+
+    // if search term is empty find the top counts for the domain
+    if (isTopCountsSearch(term)) {
+      return getTopCountsSearchWithStandard(domain, surveyName, standard, pageRequest);
+    }
+
+    Map<String, String> parsedTerms = modifyTermMatchUseEndsWithTerms(term);
+    String modifiedSearchTerm = parsedTerms.get(MODIFIED_TERMS);
+    List<String> endsWithTerms = new ArrayList<>();
+    if (parsedTerms.get(ENDS_WITH_TERMS).length() > 0) {
+      endsWithTerms.addAll(Arrays.asList(parsedTerms.get(ENDS_WITH_TERMS).split(",")));
+    }
+
+    // if the modified search term is empty and endsWithTerms is empty return an empty result
+    if (modifiedSearchTerm.length() == 0 && endsWithTerms.isEmpty()) {
+      return new CriteriaListWithCountResponse().totalCount(0L);
+    }
+
+    // if domain type is survey and modifiedSearchTerm is not empty then return search survey
+    if (isSurveyDomain(domain) && modifiedSearchTerm.length() > 0) {
+      return findSurveyCriteriaBySearchTerm(surveyName, pageRequest, modifiedSearchTerm);
+    }
+
+    List<DbCriteria> dbCriteriaList = new ArrayList<>();
+    // find a match on concept code
+    dbCriteriaList.addAll(
+        cbCriteriaDao
```

Comment:
This dao call returns a paginated list. By converting this to a list and not using page like the previous implementation, you are losing the total count of the results. This needs to be changed or we will be introducing a regression.

Reply:
Addressed this issue as per discussion